sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -98,7 +98,10 @@ object ResolveDefaultColumns {
     if (SQLConf.get.enableDefaultColumns) {
       val allowedTableProviders: Array[String] =
         SQLConf.get.getConf(SQLConf.DEFAULT_COLUMN_ALLOWED_PROVIDERS)
-          .toLowerCase().split(",").map(_.trim)
+          .toLowerCase().split(",").map(_.trim) ++
+          // We include the empty string here to allow the 'tableProvider' argument to be empty.
+          // This is only empty in tests, for example when using the InMemoryTableCatalog.
```

Comment:
@dtenedor I don't think this is a good practice. We can do one of the following:

- use a valid provider in the tests, e.g. "parquet"
- set the config for the allowed providers in the tests
- use `if(isTesting)...` here. This is not recommended though.

Reply:
@gengliangwang sounds good, I followed the second option.